app/containers/Send/Send.jsx

```diff
@@ -191,21 +208,97 @@ export default class Send extends React.Component<Props, State> {
       if (field === 'address') {
         objectToModify.address = value
       }
+
+      if (this.props.chain === 'neo3') {
+        this.attemptToCalculateN3Fees(newState)
+      }
+
       return { sendRowDetails: newState }
     })
   }
 
+  attemptToCalculateN3Fees = async (sendRowDetails: Array<Object>) => {
+    this.setState({ loading: true })
+    const sendEntries = sendRowDetails.map((row: Object) => ({
+      address: row.address,
+      amount: toNumber(row.amount.toString()),
+      symbol: row.asset,
+    }))
+
+    let shouldCalculateFees = true
+
+    // TODO: not exactly sure what the criteria should be for
+    // attempting to calculate fees
+    sendEntries.forEach(entry => {
+      if (!n3Wallet.isAddress(entry.address)) {
+        shouldCalculateFees = false
+      }
+    })
+
+    if (shouldCalculateFees) {
+      const fees = await this.props
+        .calculateN3Fees({ sendEntries })
+        .catch(() => {
+          console.warn('An error occurred attempting to calculate fees')
+        })
+      // eslint-disable-next-line
+      this.setState({
+        n3Fees: fees,
+        loading: false,
+      })
+    }
+  }
+
   calculateMaxValue = (asset: string, index: number = 0) => {
-    const { sendableAssets } = this.props
+    const { sendableAssets, chain } = this.props
+
+    const MIN_EXPECTED_GAS_FEE = 0.072
+
+    if (chain === 'neo2') {
+      if (sendableAssets[asset]) {
+        const rows = [...this.state.sendRowDetails]
+        const rowsWithAsset = rows.filter(row => row.asset === asset)
+        const existingAmounts = this.calculateRowAmounts(asset, index)
+        const decimals = this.calculateDecimals(asset)
+        const totalSendableAssets = toBigNumber(sendableAssets[asset].balance)
+        if (rowsWithAsset.length === 1 || rowsWithAsset.length === 0) {
+          return toNumber(sendableAssets[asset].balance).toFixed(decimals)
+        }
+        return minusNumber(totalSendableAssets, existingAmounts).toFixed(
+          decimals,
+        )
+      }
+      return '0'
+    }
+
     if (sendableAssets[asset]) {
       const rows = [...this.state.sendRowDetails]
       const rowsWithAsset = rows.filter(row => row.asset === asset)
       const existingAmounts = this.calculateRowAmounts(asset, index)
       const decimals = this.calculateDecimals(asset)
-      const totalSendableAssets = toBigNumber(sendableAssets[asset].balance)
+      let totalSendableAssets = toBigNumber(sendableAssets[asset].balance)
+
+      if (asset === 'GAS') {
+        const existingGasAmounts =
+          Number(this.calculateRowAmounts(asset, index)) - MIN_EXPECTED_GAS_FEE
+
+        totalSendableAssets = minusNumber(
+          totalSendableAssets,
+          MIN_EXPECTED_GAS_FEE * rowsWithAsset.length,
```

Comment:
With this being a floating point number, I'd suggest using `multiplyNumber` here & storing `MIN_EXPECTED_GAS_FEE` as a string value instead of a floating point number.

```js
const MIN_EXPECTED_GAS_FEE = '0.072'

totalSendableAssets = minusNumber(
  totalSendableAssets,
  multiplyNumber(MIN_EXPECTED_GAS_FEE, rowsWithAsset.length)
)
```